From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 26 Jan 2021 17:12:04 +0100
Subject: extensionSystem: Fix opening Extensions app from notification

Launching the app is implemented by the source's open() method, but
only external notifications are hooked up to call into the source
when no default action was provided.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1595>
(cherry picked from commit ddc2e0f4cb0046eac46a8ca7ddf53c39e386c3e4)
Origin: upstream, 3.38.4, commit:0657309ccd5517cc19fa4ae8741cde9db655aced
---
 js/ui/extensionSystem.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/ui/extensionSystem.js b/js/ui/extensionSystem.js
index 41ef98d..6b624fc 100644
--- a/js/ui/extensionSystem.js
+++ b/js/ui/extensionSystem.js
@@ -252,6 +252,8 @@ var ExtensionManager = class {
             let notification = new MessageTray.Notification(source,
                 _('Extension Updates Available'),
                 _('Extension updates are ready to be installed.'));
+            notification.connect('activated',
+                () => source.open());
             source.showNotification(notification);
         }
     }
